synthetic-dataset/image_blending.py

```diff
@@ -28,3 +31,32 @@ def basic_blending(img, smoke, offset=(0, 0), opacity=0.8):
     mask[dy:dy+smoke.shape[0], dx:dx+smoke.shape[1], :] = mask_dst
 
     return img, mask
+
+
+def poisson_blending(img, smoke, offset=(0, 0)):
+    """Add smoke on image using poisson image blending
+
+    Args:
+        img (np.array): background image
+        smoke (np.array): smoke image
+        offset (tuple, optional): smoke location offset (dy, dx). Defaults to (0, 0).
+
+    Returns:
+        np.array: result image
+        np.array: result mask
+    """
+
+    smoke_mask = smoke[:, :, 0] > 50
```

Comment:
Interesting, we should think of way to unify mask creation. I did something different on my side : https://github.com/pyronear/synthetic-dataset/pull/2/files#diff-bb17069a2e9f6a202d1f67adc120d19b82a4cef83d33f9c048e492009f6af6deR46

Reply:
I need a 1 channel image with floatt64 type so I can't use your method. Because the conversion is inside the function I'm not sure this is really an issue